website/content/docs/concepts/acl/auth-methods/oidc.mdx

```diff
@@ -74,6 +74,107 @@ Complete the login via your OIDC provider. Launching browser to:
 Your browser opens to the generated URL to complete the provider's login.
 Enter the URL manually if the browser does not automatically open.
 
+### Client assertions
+
+Also known as "[private key JWT][]", client assertions offer a more secure
+authentication mechanism compared to client secrets. The OIDC client (in this
+case, Nomad) signs a JWT with an RSA private key (or HMAC key) that the OIDC
+provider verifies with an associated public key (or the same HMAC key).
+
+Here are some partial [auth method configuration][] examples. They focus only
+on the client assertion feature; they are not complete, functional examples.
+
+#### Nomad keyring
+
+```json
+{
+  "OIDCDiscoveryURL": "https://your-keycloak-instance.com/realms/nomad",
+  "OIDCClientID": "{your-client-id}",
+  "BoundAudiences": ["{your-client-id}"],
+  "OIDCClientAssertion": {
+    "Audience": ["https://your-keycloak-instance.com/realms/nomad"],
+    "KeySource": "nomad"
+  }
+}
+```
+
+In this example for Keycloak, Nomad signs the JWT with its own internal private
+key. It sets the JWT's "kid" header as the key ID, as presented by Nomad's
+[jwks.json][] endpoint.
+
+This is arguably the most secure option, because only Nomad has the private key.
```

Comment:
What do you think of moving this intro statement before the code block?

Reply:
I like it!